### DISCUSSION TITLE
More search limiting keywords

### DISCUSSION THREAD

**Discussion by majkinetor** [1Remote/1Remote#413] [Ideas]
Currently it is not possible to show only cerrtain protocol or search within notes. Given the list limitations on such filters I propose we introduce new kyewords, some or all of:

|Keyword|Meaning|
|---|---|
| # | show only connetions having this tag (we have this)|
| &| show only connections belong to specific db|
| $ | show only connections of specific protocol|
| %| show only connectiions containing phrase in note|
| !| show only connections having this in a hostname |
|^| show only connections having this username |

Search could use startsWith as currently with tags as it is more convenient then with exact match. This would work in launcher too.

Example:

- `$rdp db` would show only RDP connections having db in name
- `$r db` would show RDP and RemoteApp connections having db in name
- `$rdp #db` show RDP connections with tag db.
- `&local $rdp #db bla` - RDP connections having bla in name with tag db and in database Local

I frequently need to show specific protocol ($ here) but then I thought that it might make sence for other things too. We definitelly need wider search box if we allow all. Not sure if we need recent history...  Search tooltip or some (i) field next to it could show quick help.

**Comment by VShawn** [comment]
Technically, it is easy to achieve.

The fact is I can hardly remember these keyword, and in the end, I become lazy and only use the most basic search instead of looking through the manual. 

Therefore, it is necessary to provide a brief and intuitive tooltip for this function.

**Reply by majkinetor** [reply]
We can just add icon next to search. Howering it or clicking it will show above table. 

![image](https://github.com/1Remote/1Remote/assets/85767/c81acbe9-f47e-45e8-9541-eb7652d473dc)

**Reply by VShawn** [reply]
and I will not support fuzzy search and highline except for name.


----

hold on what if space in a keyword?  

If there are 
 - connection name `kitty` note with `hello word` 
 - and connection named `world` note with `hellow kitty`.

then who will be hit by `%hello kitty`?

**Reply by majkinetor** [reply]
1. Since you used `%hello`, that will hit a connections with such word in a note which means both (since search is contains I beleive)
2. `kitty` without anything looks name only which eliminates connection named `world`.

> and I will not support fuzzy except name

Yes.

> hold on what if space in a keyword?

Who cares, you can use `%hello %world` for example. I would not complicate with `%'hello world'` or `%hello-world` but its on you :) 

BTW, if we go with #416 some of above becomes outdated:

- & (db)
Only if we decide that db goes into tag, which IMO probably should, as now sort works over db groups, not something you may want - if I sort by the name, and I have 10 databases, its not good UX IMO. If I have db tag, and sort by the name, I have single list sorted by name, and I can limit it to db by clicking on tag.  Not sure... like I said on #416 maybe this should be in global settings (db as system tag vs group) but I would probably rather go with tag in order not to complicate the system. Since db can have multiple words, we could create tag with spaces replaced or let the user create unique db tag himself and put a complete name in tag decription (which should be visible in the tooltip based on #401 )
- $ (protocol)
Since protocols may go into tag, there is no need for this, we have # already

Ofc. all this if we actually go with #416. 

But anyway, like I said, do not implement stuff while it is in discussion phase. Vote for it with 👍 or 👎  so I can create implementation ticket based on results of discussion :)

**Reply by VShawn** [reply]
Since there should be a lot of phrases in the note, though I think searching in notes is a bit over-kill but  if I want to search from the note, it is very likely that I search for a phrase rather than a single word, so I think searching for phrases is necessary. And `%hello %world` looks counterintuitive.

Note that connection name, database name, protocol, and username may be a phrase rather than a single word.


> But anyway, like I said, do not implement stuff 

I won't :)
